### PR TITLE
Reduce log spam and clean up EncoderValidator

### PR DESCRIPTION
**Pull request by Bond-009**
Don't write each codec on a new line.
Use a regex to match codecs in the ffmpeg output.
ex: https://regex101.com/r/bn9IOy/12/

### CODE REVIEW

File: MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs [hunk 1]
```diff
@@ -101,32 +105,30 @@ private List<string> GetDecoders(string encoderAppPath)
                 "hevc"
             };
 
-            foreach (var codec in required)
-            {
-                var srch = " " + codec + "  ";
+            var found = required.Where(x => output.IndexOf(x, StringComparison.OrdinalIgnoreCase) != -1);
```
Comment (JustAMan) [inline MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs]: This somewhat changes original search - it prepended and appended some spaces. Do you have any idea why it did so? Maybe for codecs which might have starting or ending conflicts (e.g. codec1 and codec12). Might be worth keeping this logic...

File: MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs [hunk 2]
```diff
@@ -101,32 +105,30 @@ private List<string> GetDecoders(string encoderAppPath)
...
-                    found.Add(codec);
-                }
-            }
+            _logger.LogInformation("Available decoders: {Codecs}", found);
 
             return found;
         }
 
-        private List<string> GetEncoders(string encoderAppPath)
+        private IEnumerable<string> GetEncoders(string encoderAppPath)
```
Comment (JustAMan) [inline MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs]: Cannot we reduce copy-pasteness here? The only difference between two functions should be ffmpeg option plus the list of codecs to look for.

File: MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs (hunk 1 above)
Comment (Bond-009) [reply MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs]: Your are right, there are ex: `flashsv` and `flashsv2`. Fixed this with a regex rather that a hacky whitespace addition.

File: MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs [hunk 3]
```diff
@@ -153,9 +113,43 @@ private IEnumerable<string> GetEncoders(string encoderAppPath)
...
+            }
+
+            if (string.IsNullOrWhiteSpace(output))
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var required = codec == Codec.Encoder ? requiredEncoders : requiredDecoders;
+
+            Regex regex = new Regex(@"\s\S{6}\s(?<codec>(\w|-)+)\s{3}");
```
Comment (JustAMan) [inline MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs]: I'm not versed well enough in regexps, so a comment describing this one would be good.

**Pull request merged**